src/ejabberd_sm.erl

```diff
@@ -248,8 +250,8 @@ store_info(User, Server, Resource, {Key, _Value} = KV) ->
             end
     end.
 
--spec remove_info(jid:user(), jid:server(), jid:resource(),
-                 {any(), any()}) -> ok | {error, offline}.
+-spec remove_info(jid:user(), jid:server(), jid:resource(), info_key()) ->
```

Comment:
how about the function `maybe_remove_info(jid:user(), jid:server(),info_key())` that will try to remove info_key from all the sessions associated with the user?

Reply:
I took different approach. The maybe_remove is implemented in `mod_event_pusher_push` since we also need to look inside the info to check if the node is associated withe the session before we remove it.
Also I want to keep the more specific interface as is now.